pkg/endpoint/regeneration/owner.go

```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/cilium/cilium/pkg/lock"
 	monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
 	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/proxy/logger"
```

Comment:
If this package is now self-contained enough that it doesn't pull in lots of other dependencies, then maybe it can define its own endpoint interface instead of depending on `proxy/logger`, which seems a little weird.

Reply:
Hmm yeah that makes lots of sense. Will follow up on that. @joestringer should I add an interface that has all endpoint's methods or only the ones defined in `EndpointUpdater` and `EndpointInfoSource` for now? Oh! What stop us from doing the same for the policy package? We could add as a good-first-issue, it's an easy task to do IMO.

Reply:
I think the minimal set of methods that's required is the best. And yeah it may make sense to do the same in policy package, definitely good first issues to look into. :+1:

Reply:
@joestringer https://github.com/cilium/cilium/issues/8364